src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,1323 @@
+
+        logging.debug("Call to %s was successful.\n" % url)
+        if raw_response:
+            return resp
+        else:
+            return resp.status_code
+
+    def checkMegaFieldExists(self, mega_field_id, storage_dir):
+        """
+        Check if filename complies with set allowed characters.
+        :param mega_field_id (string): name of the mega field.
+        :param storage_dir (string): path to the mega field.
+        :return (bool): True if mega field exists.
+        """
+        ASM_FILE_ILLEGAL_CHARS = r'[^a-z0-9_()-]'
+        ASM_PATH_ILLEGAL_CHARS = r'[^A-Za-z0-9/_()-]'
+        if re.search(ASM_PATH_ILLEGAL_CHARS, storage_dir):
+            logging.error("The specified storage directory contains invalid characters, cannot check if mega field "
+                          "exists (only the characters '%s' are allowed)." % ASM_FILE_ILLEGAL_CHARS[2:-1])
+            return False
+
+        if re.search(ASM_FILE_ILLEGAL_CHARS, mega_field_id):
+            logging.error("The specified mega_field_id contains invalid characters, cannot check if mega field exists"
+                          "(only the characters '%s' are allowed)." % ASM_FILE_ILLEGAL_CHARS[2:-1])
+            return False
+
+        response = self.asmApiPostCall("/scan/check_mega_field?mega_field_id=%s&storage_directory=%s" %
+                                       (mega_field_id, storage_dir), 200, raw_response=True)
+        return json.loads(response.content)["exists"]
+
+    def _startCalibration(self, *args):
+        """
+        Call start calibration loop on the ASM and subscribe _startCalibration to all the VA's so the calibration
+        loop is restarted, with updated parameters, when one of the VA's is changed.
+
+        *args is need to support self subscribing
+        """
+        self.asmApiPostCall("/scan/stop_calibration_loop", 204)
+        descanner = self._mirror_descanner
+        scanner = self._ebeam_scanner
+        mppc = self._mppc
+
+        # Subscribing to the descanner VA's
+        descanner.rotation.subscribe(self._startCalibration)
+        descanner.scanOffset.subscribe(self._startCalibration)
+        descanner.scanGain.subscribe(self._startCalibration)
+
+        # Subscribing to the scanner VA's
+        scanner.dwellTime.subscribe(self._startCalibration)
+        scanner.rotation.subscribe(self._startCalibration)
+        scanner.scanDelay.subscribe(self._startCalibration)
+        scanner.scanOffset.subscribe(self._startCalibration)
+        scanner.scanGain.subscribe(self._startCalibration)
+
+        # Subscribing to the scanner VA
+        mppc.cellCompleteResolution.subscribe(self._startCalibration)
+
+        # Assemble the calibration data and retrieve input values
+        self._calibrationParameters = self._assembleCalibrationData()
+        self.asmApiPostCall("/scan/start_calibration_loop", 204, data=self._calibrationParameters.to_dict())
+
+    def _assembleCalibrationData(self):
```

Comment:
-> _assembleCalibrationMetadata (so it is the same as _assembleMegafieldMetadata